src/execution/mod.rs

```diff
@@ -40,21 +41,34 @@ use {
     tonic::transport::Channel,
 };
 
-use super::config::ExecutionConfig;
-use super::state::tabs::sql::Query;
-use super::AppEvent;
+use crate::app::config::ExecutionConfig;
+use crate::app::state::tabs::sql::Query;
+use crate::app::AppEvent;
 
+/// Structure for executing queries either locally or remotely (via FlightSQL)
+///
+/// This context includes both:
+///
+/// 1. The configuration of a [`SessionContext`] with  various extensions enabled
+///
+/// 2. The code for running SQL queries
+///
+/// The design goals for this module are to serve as an example of how to integrate
+/// DataFusion into an application and to provide a simple interface for running SQL queries
+/// with the various extensions enabled.
+///
+/// Thus it is important (eventually) not depend on the code in the app crate
 pub struct ExecutionContext {
-    pub session_ctx: SessionContext,
```

Comment:
when I made these non pub it turned out much of it isn't used, so I removed the unused fields for now